node/src/components/block_validator.rs

```diff
@@ -202,15 +219,57 @@ where
                 });
             }
             Event::DeployMissing(deploy_hash) => {
+                info!(%deploy_hash, "request to download deploy timed out");
                 // A deploy failed to fetch. If there is still hope (i.e. other outstanding
                 // requests), we just ignore this little accident.
                 if self.in_flight.dec(&deploy_hash) != 0 {
                     return Effects::new();
                 }
 
-                // Otherwise notify everyone still waiting on it that all is lost.
+                // Flag indicating whether we've retried fetching the deploy.
+                let mut retried = false;
+
                 self.validation_states.retain(|key, state| {
                     if state.missing_deploys.contains(&deploy_hash) {
+                        match state.source() {
+                            Some(peer) => {
+                                info!(%deploy_hash, ?peer, "trying the next peer");
+                                // There's still hope to download the deploy.
+                                let (chainspec, block_timestamp) = &state.context;
+                                effects.extend(fetch_deploy(effect_builder, Arc::clone(chainspec), *block_timestamp, deploy_hash, peer));
```

Comment:
Line too long.

Reply:
`cargo fmt` ignored this.

Reply:
I know, it sometimes gives up on a whole file, if it doesn't know how to reformat some long lines.